lib/utils/patterns.ts

```diff
@@ -282,6 +283,69 @@ export function inferOrSelectDevice(preferredUuid: string) {
 		});
 }
 
+export function getTargetUuid(
+	sdk: BalenaSdk.BalenaSDK,
+	applicationOrDevice: string,
+): Bluebird<string> {
+	const appTest = validation.validateApplicationName(applicationOrDevice);
+	const uuidTest = validation.validateUuid(applicationOrDevice);
+
+	if (!appTest && !uuidTest) {
+		throw new Error(`Device or application not found: ${applicationOrDevice}`);
+	}
+
+	const isBalenaApplicationNotFound = (e: any) =>
+		e instanceof BalenaApplicationNotFound;
+
+	// if we have a definate device UUID...
+	if (uuidTest && !appTest) {
+		return sdk.models.device
```

Comment:
We can potentially skip hitting the api, if we first test for it being a long uuid

Reply:
@CameronDiver Since this is getting a target UUID, the resulting UUID should be presumed to be online... if this is wrong, I think we should remove the `is_online` filter from each API lookup in this method, so it just returns valid/existing UUIDs irrespective of their online status. Otherwise the caller will not know if they should also check for the `is_online` status once getting the result.

Reply:
Hm, yeah I know what you mean, I tried to reconcile this in ssh by keeping track of if we've done an online check by that point (for example, when going from short uuid -> long uuid, also do an online check and keep the result). This worked pretty well but it made the code a little confusing. Perhaps you're right, and we should just always do the online check after determining the target.

Reply:
I renamed the method to `getOnlineTargetUuid` so it will only return a UUID for an online device. This makes it very clear what you will get, so I think that makes it better.